### PR TITLE
feat: update learner credit plan cards to use subsidies

### PR DESCRIPTION
**Pull request by katrinan029**
# Description
1. Updates Learner Credit Plan cards to link to subsidies instead of policies.  
2. Updates license manager django url

# Test plan
1. checkout branch `knguyen2/ent-9476`
2. at the root directory, add file `webpack.dev.config.js` with the following content:
```
const { createConfig } = require('@openedx/frontend-build');

module.exports = createConfig('webpack-dev', {
  devServer: {
    allowedHosts: 'all',
    https: true,
  },
});
```
3. replace the content in `.env.development` with this info:
```
NODE_ENV='development'
PORT=18450
FEATURE_CUSTOMER_SUPPORT_VIEW='true'
ADMIN_PORTAL_BASE_URL='https://portal.stage.edx.org'
ACCESS_TOKEN_COOKIE_NAME='stage-edx-jwt-cookie-header-payload'
BASE_URL='https://localhost.stage.edx.org:18450'
FEATURE_CONFIGURATION_MANAGEMENT='true'
FEATURE_CONFIGURATION_ENTERPRISE_PROVISION='true'
FEATURE_CONFIGURATION_EDIT_ENTERPRISE_PROVISION='true'
CREDENTIALS_BASE_URL='https://credentials.stage.edx.org'
CSRF_TOKEN_API_PATH='/csrf/api/v1/token'
ECOMMERCE_BASE_URL='https://ecommerce.stage.edx.org'
ENTERPRISE_ACCESS_BASE_URL='https://enterprise-access.stage.edx.org'
LANGUAGE_PREFERENCE_COOKIE_NAME='openedx-language-preference'
LMS_BASE_URL='https://courses.stage.edx.org'
LICENSE_MANAGER_URL='https://license-manager.stage.edx.org'
LICENSE_MANAGER_DJANGO_URL='https://license-manager-internal.stage.edx.org'
SUPPORT_CONFLUENCE='https://support-tools.edx.org'
SUPPORT_CUSTOMER_REQUEST='https://support-tools.edx.org'
DISCOVERY_API_BASE_URL='https://discovery.stage.edx.org'
LOGIN_URL='https://courses.stage.edx.org/login'
LOGOUT_URL='https://courses.stage.edx.org/logout'
LOGO_URL=https://edx-cdn.org/v3/default/logo.svg
LOGO_TRADEMARK_URL=https://edx-cdn.org/v3/default/logo-trademark.svg
LOGO_WHITE_URL=https://edx-cdn.org/v3/default/logo-white.svg/
FAVICON_URL=https://edx-cdn.org/v3/default/favicon.ico
MARKETING_SITE_BASE_URL='https://stage.edx.org'
ORDER_HISTORY_URL='https://orders.stage.edx.org/orders'
REFRESH_ACCESS_TOKEN_ENDPOINT='https://courses.stage.edx.org/login_refresh'
SEGMENT_KEY=null
SITE_NAME='edX'
SUBSIDY_BASE_URL='https://enterprise-subsidy.stage.edx.org'
SUBSIDY_BASE_DJANGO_URL='https://enterprise-subsidy-internal.stage.edx.org'
USER_INFO_COOKIE_NAME='edx-user-info'
PUBLISHER_BASE_URL='https://publisher.stage.edx.org/'
APP_ID='support-tools'
MFE_CONFIG_API_URL='https://courses.stage.edx.org/api/mfe_config/v1'
```
3. run `npm run start`
4. go to link https://localhost.stage.edx.org:18450/enterprise-configuration/customers/66b5922b-a22b-4a7b-b587-d4af0378bd6f/view and verify that the Learner Credit Plan card is a subsidy. Click on the "Open in Django" button on the card and verify that you are linked to the subsidy page for that subsidy by cross referencing the uuid of the network request url with the uuid of the django page. Check that clicking on the "View budgets" button will link you to the Learner Credit Management screen in the admin portal.
5. go to link https://localhost.stage.edx.org:18450/enterprise-configuration/customers/02982884-1f1f-4103-b6d7-3a602c1afcbd/view and verify that clicking on the "Open in Django" button on the card will link you to the django admin page for that subscription. You can do so by cross referecing the uuid of the network request url with the uuid of the django page. 

https://2u-internal.atlassian.net/browse/ENT-9476

### CODE REVIEW

File: .env.development [hunk 1]
```diff
@@ -15,6 +15,7 @@ ENTERPRISE_ACCESS_BASE_URL='http://localhost:18270'
 LANGUAGE_PREFERENCE_COOKIE_NAME='openedx-language-preference'
 LMS_BASE_URL='http://localhost:18000'
 LICENSE_MANAGER_URL='http://localhost:18170'
+LICENSE_MANAGER_DJANGO_URL='http://localhost:18170'
```
Comment (marlonkeating) [inline .env.development:18]: Is there a need for a new `LICENSE_MANAGER_DJANGO_URL` variable if the value is the same as the existing `LICENSE_MANAGER_URL`?

File: src/data/services/utils.js [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+ * @param {Array} [results] Array of results.
+ * @returns Array of all results for authenticated user.
+ */
+export default async function fetchPaginatedData(url, results = []) {
+  const response = await getAuthenticatedHttpClient().get(url);
+  const responseData = camelCaseObject(response.data);
+  const resultsCopy = [...results];
+  resultsCopy.push(...responseData.results);
+  if (responseData.next) {
+    return fetchPaginatedData(responseData.next, resultsCopy);
```
Comment (marlonkeating) [inline src/data/services/utils.js:18]: Looks like we could use some test coverage in the pagination case.

File: src/Configuration/Customers/data/utils.js [hunk 3]
```diff
@@ -24,10 +26,15 @@ export const getCustomerSubscriptions = async (enterpriseId) => {
...
-  const response = await EnterpriseAccessApiService.fetchSubsidyAccessPolicies(enterpriseId);
-  const subsidyAccessPolicies = camelCaseObject(response.data);
-  return subsidyAccessPolicies;
+export const getSubsidies = async (enterpriseUUID) => {
+  const url = `${getConfig().SUBSIDY_BASE_URL}/api/v1/subsidies/?enterprise_customer_uuid=${enterpriseUUID}`;
+  try {
+    const { results } = await fetchPaginatedData(url);
+    return results;
+  } catch (error) {
+    logError(error);
```
Comment (marlonkeating) [inline src/Configuration/Customers/data/utils.js:35]: Looks like we could use some test coverage in the error case.

**Review by marlonkeating — CHANGES_REQUESTED** [review]

File: .env.development (hunk 1 above)
Comment (katrinan029) [reply .env.development:18]: yes, the `LICENSE_MANAGER_URL` variable is use for fetching subscriptions data and the `LICENSE_MANAGER_DJANGO_URL` is used as the django admin url. we ran into an issue with fetching the data when we used the same url. the fetching one should be `license-manager.edx.org` and the admin django screen is `license-manager-internal.edx.org`.

**Review by marlonkeating — APPROVED** [review]

**Pull request merged**